frame/treasury/src/lib.rs

```diff
@@ -364,6 +368,10 @@ pub mod pallet {
 		/// - O(1)
 		#[pallet::call_index(1)]
 		#[pallet::weight((T::WeightInfo::reject_proposal(), DispatchClass::Operational))]
+		#[allow(deprecated)]
+		#[deprecated(
+			note = "`reject_proposal` will be removed in February 2024. Use `spend` instead."
+		)]
 		pub fn reject_proposal(
```

Comment:
You could rename them to `deprecated_reject_proposal`. Then the info will be in the metadata and visible to downstream devs/users.

Reply:
it can be a solution actually, but I would first agree on it and have it as a standard to not create multiple ways of deprecating